File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        set<char> sett;
        int left = 0, right = 0;
        int maxlength=0 ;

        while (right < s.size()) {
            if (sett.find(s[right]) == sett.end()) {
                sett.insert(s[right]);
                maxlength = max(maxlength, right - left + 1);
                right++;
            } else {
                sett.erase(s[left]);
                left++;
            }
        }

        return maxlength;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int start=0;
        int end=0;
        int ans=0;
        int freq[]=new int[128];
        while(end<s.length()){
            freq[s.charAt(end)]++;
            while(start<=end && freq[s.charAt(end)]>=2){
                freq[s.charAt(start)]--;
                start++;
            }
            ans=Math.max(ans,end-start+1);
            end++;
        }
        return ans;
    }
}